docs/roomHandlers/fishingRoom.md

```diff
@@ -0,0 +1,41 @@
+# Fishing Room
+
+A room with a beach and a boat dock. The player can fish from dock or leave. Fishing may result in catching a fish.
+
+## States
+
+- **AtEntrance**: The player is on a dock with a fishing pole
+- **Fishing**: The player casts their fishing line in hopes to find a fish
+- **Caught**: The player now has a fish
+- **BrokenLine**: The player has a broken their fishing line and the fish got away
+- **FixLine**: The player can fix their broken line to fish again
+
+## Transitions
+
+- **leave**: The player leaves through the only available door
+- **fish**: The player casts their fishing line
+- **catch**: The player has caught a fish
+- **break**: The player lost the fish
+- **fix**: The player fixs their fishing pole so they can fish again
```

Comment:
My intention for the first iteration of these state diagrams was for transitions to equate to a user command. 

For example if you have the transition `A --> B: actionK`, and the user is in state `A` when they perform `klicker-knight actionK` then the game would output the description of `actionK` and the description of state `B`.

So given these descriptions, if the user does `klicker-knight fish` at the entrance then the game would output "You cast your fishing line. You cast your fishing line in hopes of finding a fish" which is a bit redundant.

These descriptions also imply that the user can do `klicker-knight catch` when in the fishing state. Also if they do `klicker-knight` when at the fishing state they would cancel out of fishing and return to the entrance, which I don't think is intended.

I suggest that you replace `catch` and `break` with the `fish` command. When the player does `klicker-knight fish` there's a hidden random number associated with the fish attempt. This random number would determine if the player stays in the fishing state, catches a fish or breaks the line. So your diagram would have this for the Fishing transitions:

```mermaid
  Fishing --> AtEntrance: stop
  Fishing --> Caught: fish (caught)
  Fishing --> BrokenLine: fish (broke)
  Fishing --> Fishing: fish (idle)
```

Reply:
I've updated the wording. I wasn't sure what would output when, but I think I understand that a little bit better now.

There will be some kind of random number/chance thing that will determine what happens.